app/views/good_job/shared/_filter.erb

```diff
@@ -31,12 +31,23 @@
           </select>
         </div>
 
+        <div class="me-md-2 mb-2 mb-md-0">
+          <%= label_tag "label_filter", t(".job_name"), class: "visually-hidden" %>
+          <select name="labels[]" id="label_filter" class="form-select form-select-sm" size="2" multiple>
+            <option value="_all" <%= "selected='selected'" if params[:labels].blank? %>><%= t ".all_labels" %></option>
+
+            <% filter.labels.each do |name, count| %>
+              <option value="<%= name.to_param %>" <%= "selected='selected'" if params[:labels]&.include?(name.to_param) %>><%= name %> (<%= number_with_delimiter(count) %>)</option>
+            <% end %>
+          </select>
+        </div>
+
         <div class="me-md-2 mb-2 mb-md-0 flex-fill">
           <%= label_tag "query", t(".search"), class: "visually-hidden" %>
           <%= search_field_tag "query", params[:query], class: "form-control form-control-sm", placeholder: t(".placeholder"), autocomplete: "off" %>
         </div>
 
-        <div class="d-flex flex-col align-items-end">
+        <div class="d-flex flex-col align-items-start">
```

Comment:
I did this so that the `Search` and `Clear` buttons were positioned nicely. Otherwise it would look like this:
<img width="1223" alt="image" src="https://github.com/user-attachments/assets/531a48d4-7fba-44c3-b185-ab6b71626084" />